x/reward/keeper/query_server.go

```diff
@@ -0,0 +1,203 @@
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	rewardProgram, err := k.GetRewardProgram(sdkCtx, req.GetId())
+	if err != nil {
+		return &types.QueryRewardProgramByIDResponse{}, status.Errorf(codes.Internal, fmt.Sprintf("unable to query for reward program by ID: %v", err))
+	}
+	return &types.QueryRewardProgramByIDResponse{RewardProgram: &rewardProgram}, nil
+}
+
+// returns paginated ClaimPeriodRewardDistributions
+func (k Keeper) ClaimPeriodRewardDistributions(ctx context.Context, req *types.QueryClaimPeriodRewardDistributionsRequest) (*types.QueryClaimPeriodRewardDistributionsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	response := types.QueryClaimPeriodRewardDistributionsResponse{}
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	kvStore := sdkCtx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(kvStore, types.ClaimPeriodRewardDistributionKeyPrefix)
+	pageRes, err := query.FilteredPaginate(prefixStore, req.Pagination, func(key []byte, value []byte, accumulate bool) (bool, error) {
+		var claimPeriodRewardDist types.ClaimPeriodRewardDistribution
+		vErr := claimPeriodRewardDist.Unmarshal(value)
+		if accumulate {
+			if vErr == nil {
+				response.ClaimPeriodRewardDistributions = append(response.ClaimPeriodRewardDistributions, claimPeriodRewardDist)
+			} else {
+				return false, vErr
+			}
+		}
+		return true, nil
+	})
+	if err != nil {
+		return &response, status.Error(codes.Unavailable, err.Error())
+	}
+	response.Pagination = pageRes
+	return &response, nil
+}
+
+// ClaimPeriodRewardDistributionsByID returns a ClaimPeriodRewardDistribution by rewardID and claimPeriodID
+func (k Keeper) ClaimPeriodRewardDistributionsByID(ctx context.Context, req *types.QueryClaimPeriodRewardDistributionsByIDRequest) (*types.QueryClaimPeriodRewardDistributionsByIDResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	response := types.QueryClaimPeriodRewardDistributionsByIDResponse{}
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	ClaimPeriodReward, err := k.GetClaimPeriodRewardDistribution(sdkCtx, req.GetClaimPeriodId(), req.GetRewardId())
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, fmt.Sprintf("unable to query claim period reward distributions by ID: %v", err))
+	}
+
+	if ClaimPeriodReward.Validate() == nil {
+		response.ClaimPeriodRewardDistribution = &ClaimPeriodReward
+	}
+
+	return &response, nil
+}
+
+func (k Keeper) RewardDistributionsByAddress(ctx context.Context, request *types.QueryRewardDistributionsByAddressRequest) (*types.QueryRewardDistributionsByAddressResponse, error) {
+	if request == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	address, err := sdk.AccAddressFromBech32(request.Address)
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, err.Error())
```

Comment:
The `sdkerrors.Wrap` and `Wrapf` functions are deprecated. When you've got a specific error (like here), you should use the `.Wrap` or `.Wrapf` off that error.
```suggestion
		return nil, sdkerrors.ErrInvalidAddress.Wrap(err.Error())
```
There are many other places where it's used, but I'll just leave this one comment.

Reply:
changed